File: ArrowMove/src/game/data/CommandData.java
package game.data;

import game.controller.Controller;
import game.system.Game;

public abstract class CommandData<T extends Game> extends Data {
	protected Controller controller;

	public CommandData(Controller controller) {
		this.controller = controller;
	}
	
	public abstract TextCommandData toTextCommandData();
	
	public Controller getController() {
		return controller;
	}
	
	//実行
	public abstract void execute(T game);
	//もとに戻す
	public abstract void undo(T game);
	
	
	
	

}

File: main/java/game/data/CommandData.kt
package game.data

import game.controller.Controller
import game.system.Game

abstract class CommandData(controller: Controller<*>) : Data() {
    var controller: Controller<*>
        protected set

    init {
        this.controller = controller
    }

    abstract fun toTextCommandData(): TextCommandData
//
//    //実行
//    abstract fun execute(game: T)
//
//    //もとに戻す
//    abstract fun undo(game: T)


}
